Sources/Tabs/HomeCoordinator.swift

```diff
@@ -43,16 +48,39 @@ final class HomeCoordinator: NSObject,
     }
     
     func showDeveloperMenu() {
-        let navController = UINavigationController()
-        let viewModel = DeveloperMenuViewModel()
         if tokenHolder.accessToken == nil,
            let accessToken = try? userStore.readItem(itemName: .accessToken, storage: .authenticated) {
             tokenHolder.accessToken = accessToken
         }
-        networkClient = NetworkClient(authenticationProvider: tokenHolder)
+        let networkClient = NetworkClient(authenticationProvider: tokenHolder)
+        let viewModel = DeveloperMenuViewModel { [unowned self] in
+            MainCoordinator.reauth = true
+            root.dismiss(animated: true)
+            tokenHolder.clearTokenHolder()
+            userStore.refreshStorage(accessControlLevel: LAContext().isPasscodeOnly ? .anyBiometricsOrPasscode : .currentBiometricsOrPasscode)
+            let ra = ReauthCoordinator(window: window,
+                                       analyticsService: analyticsService,
+                                       userStore: userStore)
+            openChildModally(ra, animated: true)
+            reauthCoordinator = ra
+        }
```

Comment:
Passing the action here for when the user selects to interact with the hello-world API but they're not authenticated feels weird and potentially wrong as it can't be tested effectively but seemed to be the only feasible way to do it to have it work within the coordinator pattern.

Reply:
It might make more sense to inject the HomeCoordinator to the DeveloperMenuViewController.  It's not an atypical way to use coordinators.  So something like

`weak var parentCoordinator: HomeCoordinator`

```
    private func helloWorldUnauthorizedPath() {
        if userStore.validAuthenticatedUser {
            // do stuff
        } else {
            parentCoordinator.stuffYouCurrentlyHaveAsViewModelAction()
        }
```

Reply:
Another thought:  Since in this particular case, we are ultiimately dismissing the DeveloperMenuViewController if userStore.validAuthenticatedUser is false, you could just call finish in that instance, and then handle all the clean up in performChildCleanup, etc.